pkg/execution/state/redis_state/redis_state.go

```diff
@@ -0,0 +1,349 @@
+package redis_state
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/inngest/inngest-cli/inngest"
+	"github.com/inngest/inngest-cli/pkg/execution/state"
+	"github.com/inngest/inngest-cli/pkg/execution/state/inmemory"
+	"github.com/oklog/ulid/v2"
+)
+
+const (
+	keyPrefix = "inngest:state"
+
+	// defaultExpiry is used as a placeholder while we configure open-source
+	// workflow timeouts.  Right now, data stored in redis never has a TTL.
+	defaultExpiry = 0
+)
+
+// Opt represents an option to use when creating a redis-backed state store.
+type Opt func(r *mgr)
+
+// New returns a state manager which uses Redis as the backing state store.
+//
+// TODO: Explain approach to storing state.
+func New(opts ...Opt) state.Manager {
+	m := &mgr{
+		kf: defaultKeyFunc{},
+	}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	if m.r == nil {
+		m.r = redis.NewClient(&redis.Options{
+			Addr:     "localhost:6379",
```

Comment:
We should decide on some standard env variables for this, like `<package>_<var>` and a standard pattern to have defaults either through viper or similar. I could see this being `REDIS_STATE_ADDR` or `STATE_REDIS_ADDR`. The package could be useful as one might use Redis as a queue as well which could be the same instance of Redis or different.

Reply:
Yeah, agree.  We need to document how to initialize a new self-hosted server with config stored in eg. `/etc/inngest/config.json` or via a --config path.

Reply:
https://github.com/inngest/inngest-cli/issues/124